index.js

```diff
@@ -17,54 +17,69 @@ const NativeCallDetectorAndroid = NativeModules.CallDetectionManagerAndroid
 var CallStateUpdateActionModule = require('./CallStateUpdateActionModule')
 BatchedBridge.registerCallableModule('CallStateUpdateActionModule', CallStateUpdateActionModule)
 
-const requestPermissionsAndroid = async (permissionMessage) => {
-      await PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE)
-      .then(async (gotPermission) => gotPermission
-          ? true
-          : await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE, permissionMessage)
-              .then((result) => result === PermissionsAndroid.RESULTS.GRANTED)
-        )
+// https://stackoverflow.com/questions/13154445/how-to-get-phone-number-from-an-incoming-call : Amjad Alwareh's answer.
+const requestPermissionsAndroid = (permissionMessage) => {
+  return Promise.all([
+    PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE)
+      .then((gotPermission) => gotPermission
+        ? true
+        : PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE, permissionMessage)
+          .then((result) => result === PermissionsAndroid.RESULTS.GRANTED)
+      ),
+    PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.READ_CALL_LOG)
+      .then((gotPermission) => gotPermission
+        ? true
+        : PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CALL_LOG, permissionMessage)
+          .then((result) => result === PermissionsAndroid.RESULTS.GRANTED)
+      ),
+  ])
```

Comment:
The version check should be done here. In case `Platform.constants.Release >= 9`, only a request for `READ_CALL_LOG` is needed afaik, which means this could be
```suggestion
  const requiredPermission = Platform.constants.Release >= 9
    ? PermissionsAndroid.PERMISSIONS.READ_CALL_LOG
    : PermissionsAndroid.PERMISSIONS.READ_PHONE_STATE

  return PermissionsAndroid.check(requiredPermission)
    .then((gotPermission) => gotPermission
      ? true
      : PermissionsAndroid.request(requiredPermission, permissionMessage)
        .then((result) => result === PermissionsAndroid.RESULTS.GRANTED)
    )
```

Reply:
Thanks DragonRaider! No worries about the delay. 

I have applied this fix, and updated the promise callback on line 52 to handle single promise result ( vs double promise result in previous commit ). 

My thinking was that we would need to request both permissions for android 9 to work. But I admit, I never tested to see if android 9 only needed the new permission (READ_CALL_LOG). 
Now that you have pointed this out, the code looks much much cleaner :D Let me know what you think!